frontend/src/features/profile/manage/hooks/claimVendor.ts

```diff
@@ -3,36 +3,52 @@
 import { createClient } from '@/lib/supabase/server';
 import { supabaseAdmin } from '@/lib/admin-client';
 
-export async function claimVendor(accessToken: string) {
-  // Get current user
+export async function claimVendor(accessToken: string, userId?: string) {
   const supabase = await createClient();
-  const { data: { user } } = await supabase.auth.getUser();
-  
-  if (!user) {
-    throw new Error('User not authenticated');
+
+  // If userId is provided (for new signups), use that; otherwise get from session
```

Comment:
how will userId be provided in new sign ups?

Reply:
now that we're using createUser, it'll be never be null! removing